// @stackframe/stack-shared/dist/interface/crud/transactions.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/interface/crud/transactions.ts
var transactions_exports = {};
__export(transactions_exports, {
  adminTransaction: () => adminTransaction
});
module.exports = __toCommonJS(transactions_exports);
var import_schema_fields = require("../../schema-fields.js");
var adminTransaction = (0, import_schema_fields.yupObject)({
  id: (0, import_schema_fields.yupString)().defined(),
  type: (0, import_schema_fields.yupString)().oneOf(["subscription", "one_time", "item_quantity_change"]).defined(),
  created_at_millis: (0, import_schema_fields.yupNumber)().defined(),
  customer_type: import_schema_fields.customerTypeSchema.defined(),
  customer_id: (0, import_schema_fields.yupString)().defined(),
  quantity: (0, import_schema_fields.yupNumber)().defined(),
  test_mode: (0, import_schema_fields.yupBoolean)().defined(),
  product_display_name: (0, import_schema_fields.yupString)().nullable().defined(),
  price: import_schema_fields.productPriceSchema.omit(["serverOnly", "freeTrial"]).nullable().defined(),
  status: (0, import_schema_fields.yupString)().nullable().defined(),
  item_id: (0, import_schema_fields.yupString)().optional(),
  description: (0, import_schema_fields.yupString)().nullable().optional(),
  expires_at_millis: (0, import_schema_fields.yupNumber)().nullable().optional()
}).defined();
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  adminTransaction
});
